jzp-code/kukdo/project/cloud-storage/cloud-storage-parent/cloud-storage-web/src/main/java/com/xiaomi/service/impl/AliyunStorageServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.xiaomi.service.impl;
+
+import com.aliyun.oss.ClientConfiguration;
+import com.aliyun.oss.OSS;
+import com.aliyun.oss.OSSClientBuilder;
+import com.xiaomi.service.StorageService;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.InputStream;
+
+@Service("AliyunStorage")
+public class AliyunStorageServiceImpl implements StorageService {
+
+    @Value("${aliyun.oss.endpoint}")
+    private String endpoint;
+
+    @Value("${aliyun.oss.accessKeyId}")
+    private String accessKeyId;
+
+    @Value("${aliyun.oss.accessKeySecret}")
+    private String accesKeySecret;
+
+    @Value("${aliyun.oss.bucketName}")
+    private String bucketName;
+
+    private OSS getOSSClient() {
+        ClientConfiguration conf = new ClientConfiguration();
+//        conf.setConnectionTimeout(5000);
+//        conf.setSocketTimeout(10000);
+//        conf.setMaxErrorRetry(3);
+        return new OSSClientBuilder().build(endpoint, accessKeyId, accesKeySecret);
+    }

```

Comment:
we need not creating oss clients every time we use it. it can be a singleton and its lifecycle should be the same as that of the application.

Reply:
> we need not creating oss clients every time we use it. it can be a singleton and its lifecycle should be the same as that of the application.

Thanks for advice, I created a new `OssClientConfig` configuration class to manage instances of OSSClient. This class will configure the OSSClient as a Spring bean so that it remains singleton throughout the application lifecycle.